src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -388,5 +394,18 @@ internal static void ThrowInvalidOperation_HasToBeArrayClass()
             throw new InvalidOperationException();
 #endif
         }
+
+        internal static bool IsMaxDepthExceeded(TypeNameParseOptions options, int depth)
+#if SYSTEM_PRIVATE_CORELIB
+            => false; // CoreLib does not enforce any limits
+#else
+            => depth > options.MaxNodes;
```

Comment:
important: so far the check was greater of equal (`>=`). After wrapping my head around I got the conclusion that we should allow for the max value too (it feels more natural).